runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -232,20 +254,53 @@ public void updateStreamTails(long address, ILogData logData) {
      * @param entry entry to potentially apply
      * @return if we need to apply the entry.
      */
-    private boolean shouldEntryBeApplied(UUID streamId, SMREntry entry) {
+    private boolean shouldEntryBeApplied(UUID streamId, SMREntry entry, boolean isCheckpointEntry) {
+        // If white list mode, ignore everything that is not in the list or checkpoints
+        if (whiteList && !streamsToLoad.contains(streamId)) {
+            return false;
+        }
+
         // We ignore the transaction stream ID because it is a raw stream
         // We don't want to create a Map for it.
         if (streamId == runtime.getObjectsView().TRANSACTION_STREAM_ID || streamsToIgnore.contains(streamId)) {
             return false;
         }
 
-        if (!streamsMetaData.containsKey(streamId) || entry.getEntry().getGlobalAddress() >
+        if (isCheckpointEntry || !streamsMetaData.containsKey(streamId) || entry.getEntry().getGlobalAddress() >
```

Comment:
Just use static boolean isCheckPointEntry(ILogData logData) ?

Reply:
I am too deep for that. I don't have access to the logData directly (or I would need to get it backward from the logEntry). Here I am processing an entry inside the CheckpointEntry. Easiest way to know if we are processing a checkpoint or a normal entry is to pass it along the parameters.